.github/workflows/python-package.yml

```diff
@@ -9,58 +9,77 @@ on:
+                      python-version: '3.12'
+                      numpy-version: '<2.3'
+                      astropy-version: '<7.1'
+                    - os: ubuntu-latest
+                      python-version: '3.11'
+                      numpy-version: '<2.1'
+                      astropy-version: '<7'
+                    # Similar to NERSC but with last NumPy < 2.
+                    - os: ubuntu-latest
+                      python-version: '3.10'
+                      numpy-version: '<2'
+                      astropy-version: '<6.1'
 
         steps:
             - name: Checkout code
-              uses: actions/checkout@v2
-              with:
-                fetch-depth: 0
+              uses: actions/checkout@v4
             - name: Set up Python ${{ matrix.python-version }}
-              uses: actions/setup-python@v2
+              uses: actions/setup-python@v5
               with:
                 python-version: ${{ matrix.python-version }}
             - name: Install Python dependencies
               run: |
-                python -m pip install --upgrade pip wheel
-                python -m pip install --upgrade "numpy${{ matrix.numpy-version }}"
-                python -m pip install --upgrade "astropy${{ matrix.astropy-version }}"
+                python -m pip install --upgrade pip setuptools wheel
+                python -m pip install --upgrade --upgrade-strategy only-if-needed "numpy${{ matrix.numpy-version }}"
+                python -m pip install --upgrade --upgrade-strategy only-if-needed "astropy${{ matrix.astropy-version }}"
                 python -m pip install --editable .[test]
             - name: Run the test
               run: pytest
 
+    # This test setup is very similar to NERSC.
     coverage:
         name: Test coverage
         runs-on: ${{ matrix.os }}
         strategy:
             fail-fast: false
             matrix:
                 os: [ubuntu-latest]
-                python-version: ['3.9']
+                python-version: ['3.10']
+                numpy-version: ['<1.23']
                 astropy-version: ['<5.1']
 
         steps:
             - name: Checkout code
-              uses: actions/checkout@v2
-              with:
-                fetch-depth: 0
+              uses: actions/checkout@v4
             - name: Set up Python ${{ matrix.python-version }}
-              uses: actions/setup-python@v2
+              uses: actions/setup-python@v5
               with:
                 python-version: ${{ matrix.python-version }}
             - name: Install Python dependencies
               run: |
-                python -m pip install --upgrade pip wheel
+                python -m pip install --upgrade pip setuptools wheel
+                python -m pip install --upgrade --upgrade-strategy only-if-needed "numpy${{ matrix.numpy-version }}"
+                python -m pip install --upgrade --upgrade-strategy only-if-needed "astropy${{ matrix.astropy-version }}"
```

Comment:
Admittedly feature creep, but ... it has always seemed a maintenance hassle that the coverage test configuration is so similar to one of the standard test matrix configurations, and when we have to change stuff like "--upgrade-strategy only-if-needed" we effectively have to do it in multiple places.  Is there any syntax option to support if/then logic so that the coverage check could piggy-back on one of the normal test matrix configurations?  i.e. run "pytest" by default, but for one of the configs run "pytest --cov"?  Then we only have to maintain the numpy, fitsio, etc. installation details in one place.